### PR TITLE
iOS workflow support

### PR DESCRIPTION
**Pull request by fritzlb**
updated build.yml to support building for iOS. Butterfly on iOS can be sideloaded using altstore (or sth similar) but is not stable at all atm, things like trying to access the camera result in a crash.

Partially solves #244

### CODE REVIEW

**Review by CodeDoctorDE — CHANGES_REQUESTED** [review]
Thanks for contributing.

It would be nice if you could add a docs/community/download/ios.md file and link it on the download site (you can copy the content of macos.md). I would only release nightly versions with it. Here it would be cool if you can add a guide how to install it.
On this download page you could link to the issue where we can post new updates.

File: app/ios/Runner/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 1]
```diff
@@ -1,122 +1,14 @@
 {
   "images" : [
```
[the remark this answers is not in the document]
Comment (fritzlb) [reply app/ios/Runner/Assets.xcassets/AppIcon.appiconset/Contents.json:2]: I only removed the iOS app icon images that show the flutter logo, I replaced it with one 1024x1024 butterfly icon which seems to be enough for iOS.

[the remark this answers is not in the document]
Comment (CodeDoctorDE) [reply app/ios/Runner/Assets.xcassets/AppIcon.appiconset/Contents.json:2]: oh, i didn't saw it. thanks

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -361,6 +361,48 @@ jobs:
...
+          flutter build ios --release --no-codesign
+      - name: Packing IPA
+        run: |
+          cd build/ios/iphoneos/
+          rm -rf Payload
+          mkdir Payload
+          cp -R Runner.app Payload/
+          rm -f linwood-butterfly-iOS.ipa
+          zip -vr linwood-butterfly-iOS.ipa Payload/
+          ls -l linwood-butterfly-iOS.ipa
```
Comment (CodeDoctorDE) [inline .github/workflows/build.yml:398]: it needs also be changed here, i would change it manually

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -361,6 +361,48 @@ jobs:
...
+          flutter-version: ${{ env.FLUTTER_VERSION }}
+          channel: 'master'
+      - name: 📦 Get dependencies
+        run: |
+          flutter clean
+          flutter pub get
+          flutter doctor -v
+      - name: 🏭 Build
+        run: |
+          flutter build ios --release --no-codesign
```
[the remark this answers is not in the document]
Comment (fritzlb) [reply .github/workflows/build.yml]: I changed that 👍

File: .github/workflows/build.yml (hunk 2 above)
Comment (fritzlb) [reply .github/workflows/build.yml]: ✅

**Comment by fritzlb** [comment]
I added iOS.md, the minimum target is 12.0 and updated the iOS building instructions. Is there anything else that needs to be done?

**Comment by CodeDoctorDE** [comment]
If you add the download-artifact and the binary to deploy, I can merge it.
On the ios download page, the file name is wrong, it needs to be changed to `linwood-butterfly-ios.ipa`.

Do you want to add the [Uniform Type Identifiers](https://developer.apple.com/documentation/uniformtypeidentifiers) to all `XTypeGroup`s (you can search for this in the code)?
Otherwise I will try to add it after merging the pr

**Comment by CodeDoctorDE** [comment]
ios.md has an error:
```
Cause: Unexpected closing slash `/` in tag, expected an open tag first
```

**Comment by fritzlb** [comment]
the deployment section should be fixed by now, I'd probably have to set up sth to test it.
the links in ios.md are fixed, note to self: just because a browser correctly displays <a href></a> doesn't mean I should use it in markdown.

I have pretty much no experience with uniform type identifiers so I'd be cool if you could add them :)
Also thank you so much for your feedback, in the last two days I learned more about github actions than ever before :)

**Comment by CodeDoctorDE** [comment]
Thanks for your contribution!
Without your help, there would be no iOS support.
I also need to research a bit. On the file selector pub package, it says it needs to have it to work on iOS.
I will merge it this evening. Feel free to contribute more.
There is another ci issue currently opened about a wrong nightly check.

**Review by CodeDoctorDE — APPROVED** [review]
thanks, looks good to me!

**Pull request merged**